modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1373 @@
+        when(AbstractResourceManager.encodeSCIMException(any(CharonException.class)))
+                .thenThrow(CharonException.class);
+
+        when(userManager.getMe(existingId,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenReturn(userOld);
+
+        when(userManager.updateMe(anyObject(), anyObject())).thenReturn(null);
+
+        //Assertions
+        Assert.assertThrows(CharonException.class, () -> {
+            meResourceManager.updateWithPATCH(existingId, scimObjectString,
+                    userManager, attributes, excludeAttributes);
+        });
+    }
+
+    //BadRequestException
+    @Test(dataProvider = "dataForUpdateWithPATCHError")
+    public void testUpdateWithPATCHBadRequestException(String existingId, String scimObjectString,
+                                                       String attributes, String excludeAttributes,
+                                                       Object objectNEWUser,
+                                                       Object objectOLDUser)
+            throws BadRequestException, NotFoundException, CharonException, NotImplementedException {
+
+        User userNew = (User) objectNEWUser;
+        User userOld = (User) objectOLDUser;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn("https://localhost:9443/scim2/Me");
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                .thenThrow(BadRequestException.class);
+
+        when(userManager.getMe(existingId,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenThrow(BadRequestException.class);
+
+        //Assertions
+        Assert.assertThrows(BadRequestException.class, () -> {
+            meResourceManager.updateWithPATCH(existingId, scimObjectString,
+                    userManager, attributes, excludeAttributes);
+        });
+    }
+
+    //NotImplementedException
+    @Test(dataProvider = "dataForUpdateWithPATCHError")
+    public void testUpdateWithPATCHNotImplementedException(String existingId, String scimObjectString,
+                                                           String attributes, String excludeAttributes,
+                                                           Object objectNEWUser,
+                                                           Object objectOLDUser)
+            throws BadRequestException, NotFoundException, CharonException, NotImplementedException {
+
+        User userNew = (User) objectNEWUser;
+        User userOld = (User) objectOLDUser;
```

Comment:
refer: https://github.com/wso2/charon/pull/328/files#r634332241

Reply:
Fixed in commit b3385c2ce740925ccf69bcb3792cd019b0c49b30